File: src/main/java/ru/ifmo/is/mfl/movies/dto/MovieDto.java
package ru.ifmo.is.mfl.movies.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;
import ru.ifmo.is.mfl.common.framework.dto.CrudDto;

import java.time.LocalDate;

@Data
@EqualsAndHashCode(callSuper = true)
public class MovieDto extends CrudDto {
  private int id;
  private String title;
  private String description;
  private String poster;
  private LocalDate releaseDate;
  private Integer duration;
  private String categories;
  private String tags;
  private String productionCountry;
  private String genres;
  private String actors;
  private String director;
  private Integer seasons;
  private Integer series;
}

File: frontend/src/interfaces/movie/dto/MovieDto.ts
import { CrudDto } from "@/interfaces/crud/dto/CrudDto";
import { LocalDate } from "@/types/LocalDate";

export interface MovieDto extends CrudDto {
  title: string;
  description: string;
  poster: string;
  releaseDate?: LocalDate;
  duration?: number;
  rating?: number;
  categories?: string;
  tags?: string;
  productionCountry?: string;
  genres?: string;
  actors?: string;
  director?: string;
  seasons?: number;
  series?: number;
}